tfe/tool_helpers.go

```diff
@@ -19,5 +20,37 @@ func fetchTerraformVersionID(version string, client *tfe.Client) (string, error)
 		return "", fmt.Errorf("terraform version not found")
 	}
 
-	return versions.Items[0].ID, nil
+	if versions.Items[0].Version != version {
+
+		options := tfe.AdminTerraformVersionsListOptions{}
+
+	found:
+		for {
+			vlist, err := client.Admin.TerraformVersions.List(ctx, options)
+			if err != nil {
+				return "", fmt.Errorf("error reading Terraform Versions: %w", err)
+			}
+
+			if vlist.CurrentPage >= vlist.TotalPages {
+				break
+			}
+
+			for _, v := range vlist.Items {
+				if v.Version == version {
+					versionID = v.ID
+					break found
+				}
+			}
+
+			options.PageNumber = vlist.NextPage
+		}
+	} else {
+		versionID = versions.Items[0].ID
+	}
+
+	if versionID == "" {
+		return "", fmt.Errorf("terraform version not found")
+	}
+
+	return versionID, nil
```

Comment:
You _could_ structure this to linear search all results without re-requesting the first page:

```suggestion
	if len(versions.Items) > 1 {
		options := tfe.AdminTerraformVersionsListOptions{}

	found:
		for {
		        for _, v := range versions {
		                if v.Version == version {
		                        versionID = v.ID
		                        break found
		                }
		        }
		        if versions.CurrentPage >= versions.TotalPages {
				break
			}
			options.PageNumber = versions.NextPage
			versions, err = client.Admin.TerraformVersions.List(ctx, options)
			
			if err != nil {
				return "", fmt.Errorf("error reading Terraform Versions: %w", err)
			}
		}
	} else if versions.Items[0].Version == version {
		versionID = versions.Items[0].ID
	}
	
	if versionID == "" {
		return "", fmt.Errorf("terraform version not found")
	}

	return versionID, nil
```

Reply:
(I didn't compile or test this code)